MANIFEST.in

```diff
@@ -6,4 +6,6 @@ include tox.ini
 include requirements.txt
 include .coveragerc
 include .codecov.yml
+include mypy.ini
 recursive-include tests *.py
+recursive-include scripts-dev *
```

Comment:
I guess these come from the template, but I'm surprised we're including them?

Reply:
I nicked that from Synapse (and this CI has a check to ensure that everything in the VCS is in the sdist. I'm not super clear on what the best practices are but I felt that either way was fine for this particular script).

Reply:
We should probably re-evaluate it, but my understanding is you usually don't include tests or random dev things.

Reply:
That doesn't entirely match my understanding.

https://discuss.python.org/t/purpose-of-an-sdist/4639/5 says you *should* include tests, since they can be used for validation, for example to verify a package passes tests on a weird system (which may even be airgapped).
The Python source code itself (essentially a source distribution) includes and runs tests, so I think it's credible that this interpretation seems conventional.

However development scripts that only make sense in a living repository indeed seem like they probably don't belong there. 

I think the idea behind the lint isn't that everything shouldn't be included, but rather that you should be conscious and explicit about which ones go in or out.
I will change it to ignore scripts-dev — the main reason I didn't do so before was that Synapse doesn't do that.

Reply:
> discuss.python.org/t/purpose-of-an-sdist/4639/5 says you should include tests, since they can be used for validation, for example to verify a package passes tests on a weird system (which may even be airgapped).

Yes that makes sense, but I've never figured out how to not have them end up in the installed package this way, if all packages do this then they get stomped on and the module makes no sense. Maybe I'm misunderstanding what this line is doing though?

>  The Python source code itself (essentially a source distribution) includes and runs tests, so I think it's credible that this interpretation seems conventional.

I'm not sure how what Python does is really relevant to packages.

Reply:
> if all packages do this then they get stomped on

How do you mean?

> I'm not sure how what Python does is really relevant to packages.

Why do you think it wouldn't be? It seems like the same principle (and Python is but one example); it's all just software packaging after all. I would imagine Debian source packages (as another example) also do this in general.

Reply:
> > if all packages do this then they get stomped on
> 
> How do you mean?

If you have two packages which include their tests and then install them you end up with a weird jumble in your `site-packages` directory (which is where your packages get installed via `pip install`).

Reply:
> If you have two packages which include their tests and then install them you end up with a weird jumble in your site-packages directory (which is where your packages get installed via pip install).

Oh, that's a different thing. That's if the package declares it as a package to be installed. It's a separate thing from it being in the source distribution, though (as an analogy, when you install Python itself with `make (alt)install`, you don't install the tests along with it).

I did verify this to be sure (can never be too careful with Python packaging... it gets more confusing each time one looks at it): if you install Synapse, you don't get any `test`/`tests` directories in your `site-packages`. I don't see how any packages could walk over each other in this case (N.B. trying to import `tests` also fails as you'd hope/expect).

This is assuming you don't install using `-e` because that doesn't do source distributions as far as I understand and doesn't seem to care about whether your files are listed in a manifest or not (or at least, as far as I can tell — I've had enough breakage in 'production' from things that were missing from the MANIFEST).